config/opal_setup_wrappers.m4

```diff
@@ -498,24 +683,299 @@ AC_DEFUN([OPAL_SETUP_WRAPPER_FINAL],[
-           [Additional CXXFLAGS_PREFIX to pass through the wrapper compilers])
-       AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_FCFLAGS, "$OMPI_WRAPPER_EXTRA_FCFLAGS",
-           [Additional FCFLAGS to pass through the wrapper compilers])
-       AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_FCFLAGS_PREFIX, "$OMPI_WRAPPER_EXTRA_FCFLAGS_PREFIX",
-           [Additional FCFLAGS to pass through the wrapper compilers])
-       AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_LDFLAGS, "$OMPI_WRAPPER_EXTRA_LDFLAGS",
-           [Additional LDFLAGS to pass through the wrapper compilers])
-       AC_DEFINE_UNQUOTED(WRAPPER_EXTRA_LIBS, "$OMPI_WRAPPER_EXTRA_LIBS",
-           [Additional LIBS to pass through the wrapper compilers])
+       AC_DEFINE_UNQUOTED(OMPI_WRAPPER_CFLAGS, "$OMPI_WRAPPER_CFLAGS",
+           [CFLAGS to pass through the wrapper compilers])
+       AC_DEFINE_UNQUOTED(OMPI_WRAPPER_CXXFLAGS, "$OMPI_WRAPPER_CXXFLAGS",
+           [CXXFLAGS to pass through the wrapper compilers])
+       AC_DEFINE_UNQUOTED(OMPI_WRAPPER_FCFLAGS, "$OMPI_WRAPPER__FCFLAGS",
+           [FCFLAGS to pass through the wrapper compilers])
+       AC_DEFINE_UNQUOTED(OMPI_WRAPPER_LDFLAGS, "$OMPI_WRAPPER_LDFLAGS",
+           [LDFLAGS to pass through the wrapper compilers])
+       AC_DEFINE_UNQUOTED(OMPI_WRAPPER_LIBS, "$OMPI_WRAPPER_LIBS",
+           [LIBS to pass through the wrapper compilers])
+
+       dnl ####################################################################
+       dnl  Setup variables for pkg-config files
+       dnl
+       dnl Add all our dependent libraries to libs.Private for users that want
+       dnl to static build, unless we're only building static libraries, in
+       dnl which case, add the dependent libraries to libs itself, since any
+       dnl linking will require the full set of libraries.
+       dnl ####################################################################
+       AC_MSG_CHECKING([for OMPI pkg-config Cflags])
+       OMPI_PC_CFLAGS="${OMPI_WRAPPER_CPPFLAGS} ${OMPI_WRAPPER_CFLAGS} ${OMPI_WRAPPER_CFLAGS_PREFIX}"
+       OMPI_PC_CFLAGS=`echo ${OMPI_PC_CFLAGS} | sed -e 's/@{/\${/g'`
+       AC_SUBST([OMPI_PC_CFLAGS])
+       AC_MSG_RESULT([${OMPI_PC_CFLAGS}])
+
+       AC_MSG_CHECKING([for OMPI pkg-config Libs])
+       OMPI_PC_LIBS="${OMPI_WRAPPER_LDFLAGS} ${OMPI_WRAPPER_LIBS}"
+       OMPI_PC_LIBS=`echo ${OMPI_PC_LIBS} | sed -e 's/@{/\${/g'`
+       AC_SUBST([OMPI_PC_LIBS])
+       AC_MSG_RESULT([${OMPI_PC_LIBS}])
+
+       AC_MSG_CHECKING([for OMPI pkg-config Libs.private])
+       OMPI_PC_LIBS_PRIVATE="${OMPI_WRAPPER_LDFLAGS_STATIC} ${OMPI_WRAPPER_LIBS_STATIC}"
+       OMPI_PC_LIBS_PRIVATE=`echo ${OMPI_PC_LIBS_PRIVATE} | sed -e 's/@{/\${/g'`
+       AC_SUBST([OMPI_PC_LIBS_PRIVATE])
+       AC_MSG_RESULT([${OMPI_PC_LIBS_PRIVATE}])
+
+       AC_MSG_CHECKING([for OMPI pkg-config Fortran Cflags])
+       OMPI_PC_CFLAGS="${OMPI_WRAPPER_FCFLAGS} ${OMPI_WRAPPER_FCFLAGS_PREFIX}"
```

Comment:
Should this be `OMPI_PC_FC_FLAGS` in this block?

Reply:
Yup.  Fixed.